File: algorithm-case/sort/bubble.js
const bubbleSort = (arrayList, asc = true) => {
  for (let i = 0; i < arrayList.length; i++) {
    for (let j = 0; j < arrayList.length - i - 1; j++) {
      if (arrayList[j] > arrayList[j + 1] && asc) {
        const temp = arrayList[j];
        arrayList[j] = arrayList[j + 1];
        arrayList[j + 1] = temp;
      }

      if (arrayList[j] < arrayList[j + 1] && !asc) {
        const temp = arrayList[i];
        arrayList[j] = arrayList[j + 1];
        arrayList[j + 1] = temp;
      }
      console.log(`process i=${i} j=${j}: ${arrayList}`);
    }
  }
  return arrayList;
};

const assertAsc = [1, 2, 3, 4, 5, 5];
// const assertDesc = [5, 5, 4, 3, 2, 1];
const random = [3, 5, 2, 1, 5, 4];
const resultAsc = bubbleSort([].concat(random));
// const resultDes = bubbleSort(random, false);
console.log(`randomElement: ${random}`);
console.log(`resultElement: ${resultAsc}`);
console.log(`Sorting result: ${assertAsc.toString() === resultAsc.toString()}`);

File: algorithm-case/sort/bubble.py
def bubbleSort(arrayList, asc=True):
    for i in range(0, len(arrayList)):
        for j in range(0, len(arrayList) - i - 1):
            if (arrayList[j] > arrayList[j+1] and asc):
                temp = arrayList[j]
                arrayList[j] = arrayList[j+1]
                arrayList[j+1] = temp
            if (arrayList[j] < arrayList[j+1] and not asc):
                temp = arrayList[j]
                arrayList[j] = arrayList[j+1]
                arrayList[j+1] = temp
    return arrayList


assertAsc = [1, 2, 3, 4, 5, 5]
# assertDes = [5, 5, 4, 3, 2, 1]
random = [3, 5, 2, 1, 5, 4]
resultAsc = bubbleSort(random.copy())
# resultDes = bubbleSort(random.copy(), False)
print('randomElement: ', random)
print('resultElement: ', resultAsc)
print('Sorting result: ', assertAsc == resultAsc)
